test/integration/generated/httpInfrastructureRest/src/clientDefinitions.ts

```diff
@@ -0,0 +1,1922 @@
+  ): Promise<
+    HttpRedirectsPatch302302Response | HttpRedirectsPatch302defaultResponse
+  >;
+  /** Post true Boolean value in request returns 303.  This request should be automatically redirected usign a get, ultimately returning a 200 status code */
+  post303(
+    options?: HttpRedirectsPost303Parameters
+  ): Promise<
+    | HttpRedirectsPost303200Response
+    | HttpRedirectsPost303303Response
+    | HttpRedirectsPost303defaultResponse
+  >;
+  /** Redirect with 307, resulting in a 200 success */
+  head307(
+    options?: HttpRedirectsHead307Parameters
+  ): Promise<
+    | HttpRedirectsHead307200Response
+    | HttpRedirectsHead307307Response
+    | HttpRedirectsHead307defaultResponse
+  >;
+  /** Redirect get with 307, resulting in a 200 success */
+  get307(
+    options?: HttpRedirectsGet307Parameters
+  ): Promise<
+    | HttpRedirectsGet307200Response
+    | HttpRedirectsGet307307Response
+    | HttpRedirectsGet307defaultResponse
+  >;
+  /** options redirected with 307, resulting in a 200 after redirect */
+  options307(
+    options?: HttpRedirectsOptions307Parameters
+  ): Promise<
+    | HttpRedirectsOptions307200Response
+    | HttpRedirectsOptions307307Response
+    | HttpRedirectsOptions307defaultResponse
+  >;
+  /** Put redirected with 307, resulting in a 200 after redirect */
+  put307(
+    options?: HttpRedirectsPut307Parameters
+  ): Promise<
+    | HttpRedirectsPut307200Response
+    | HttpRedirectsPut307307Response
+    | HttpRedirectsPut307defaultResponse
+  >;
+  /** Patch redirected with 307, resulting in a 200 after redirect */
+  patch307(
+    options?: HttpRedirectsPatch307Parameters
+  ): Promise<
+    | HttpRedirectsPatch307200Response
+    | HttpRedirectsPatch307307Response
+    | HttpRedirectsPatch307defaultResponse
+  >;
+  /** Post redirected with 307, resulting in a 200 after redirect */
+  post307(
+    options?: HttpRedirectsPost307Parameters
+  ): Promise<
+    | HttpRedirectsPost307200Response
+    | HttpRedirectsPost307307Response
+    | HttpRedirectsPost307defaultResponse
+  >;
+  /** Delete redirected with 307, resulting in a 200 after redirect */
+  delete307(
+    options?: HttpRedirectsDelete307Parameters
+  ): Promise<
+    | HttpRedirectsDelete307200Response
+    | HttpRedirectsDelete307307Response
+    | HttpRedirectsDelete307defaultResponse
+  >;
+}
+
+/** Contains operations for HttpClientFailure operations */
+export interface HttpClientFailureOperations {
+  /** Return 400 status code - should be represented in the client as an error */
+  head400(options?: HttpClientFailureHead400Parameters): Promise<HttpResponse>;
+  /** Return 400 status code - should be represented in the client as an error */
+  get400(options?: HttpClientFailureGet400Parameters): Promise<HttpResponse>;
+  /** Return 400 status code - should be represented in the client as an error */
+  options400(
+    options?: HttpClientFailureOptions400Parameters
+  ): Promise<HttpResponse>;
```

Comment:
@joheredi this is the reason why I need to add the import logic of HttpResponse in clientDefinitionGenerator. I checked the swagger, seems like they only have default response, and the llc code generator will just use HttpResponse in that case.

Reply:
Discussed offline, I think we should be generating `Promise<ErrorModelOutput>`

Reply:
This should be resolved now